### PR TITLE
61 installation and packaging

### PR DESCRIPTION
**Pull request by ognjen-j**
This PR should probably be merged last before the 1.0 release, since a lot of its functionality depends on other branches being merged first (e.g. isort tests).
TODO: add accounts to enable uploads to external repositories

### CODE REVIEW

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python3
+          export PATH=$PATH:$HOME/.poetry/bin
+          source $HOME/.poetry/env
+          virtualenv venv --python=python3
+          $HOME/.poetry/bin/poetry self:update --preview
+      - name: Install esque requirements
+        run: |
+          source ./venv/bin/activate
+          $HOME/.poetry/bin/poetry install -vvv
+      - name: Setup default esque configuration
```
Comment (swenzel) [inline .github/workflows/ci-workflow.yml]: You don't need this step, esque test suite will use its own configuration from `tests/test_configs/integration_test_config.yaml`

File: .github/workflows/ci-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          mkdir -p ~/.esque
+          source ./venv/bin/activate
+          cp ./esque/config/sample_config.yaml ~/.esque/esque.yaml
+      - name: Run unit tests
+        run: |
+          source ./venv/bin/activate
+          $HOME/.poetry/bin/poetry run pytest tests/unit/
+      - name: Run kafka stack
+        run: |
+          docker-compose up -d
```
Comment (swenzel) [inline .github/workflows/ci-workflow.yml]:
```suggestion
          docker-compose up -d
          scripts/wait-for-it.sh localhost:9092
          scripts/wait-for-it.sh localhost:2181
          scripts/wait-for-it.sh localhost:8081
```

File: .github/workflows/ci-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      - name: Run integration tests
+        run: |
+          sudo netstat -punta
```
Comment (swenzel) [inline .github/workflows/ci-workflow.yml]:
```suggestion
```

File: .github/workflows/ci-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          source ./venv/bin/activate
+          esque ctx local
```
Comment (swenzel) [inline .github/workflows/ci-workflow.yml]: Not necessary, see above.
```suggestion
```

File: .github/workflows/ci-workflow.yml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+          $HOME/.poetry/bin/poetry run pytest -s -v --integration --local tests/integration/commands/
```
Comment (swenzel) [inline .github/workflows/ci-workflow.yml]:
```suggestion
          $HOME/.poetry/bin/poetry run -- pytest -s -v --integration --local tests/integration
```

File: .github/workflows/ci-workflow.yml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      - name: Setup python and install packages
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+          architecture: x64
+        run: |
+          sudo apt-get install -y python3-setuptools python3-virtualenv python3-pip virtualenv python3-venv
+          pip3 install pre-commit
+      - name: Install and configure poetry
+        run: |
```
Comment (swenzel) [inline .github/workflows/ci-workflow.yml]: I'd rather use `pip3 install "poetry==1.0.0b3"` so we know which version we get.

File: .github/workflows/ci-workflow.yml (hunk 5 above)
Comment (ognjen-j) [reply .github/workflows/ci-workflow.yml]: Keeping it for now, so that the pipelines finish faster while I'm still testing.

**Review by hfjn — APPROVED** [review]

**Review by swenzel — APPROVED** [review]

**Pull request merged**